infrastructure/k8s/nats.tf

```diff
@@ -0,0 +1,250 @@
+resource "kubernetes_namespace" "nats" {
+  metadata {
+    name = "nats"
+  }
+}
+
+resource "null_resource" "setup_nats" {
+  triggers = {
+    nats_namespace = kubernetes_namespace.nats.metadata[0].name
+  }
+
+  provisioner "local-exec" {
+    command = <<-EOT
+      helm repo add nats https://nats-io.github.io/k8s/helm/charts/
+      helm repo update
+      helm pull nats/nats --untar
+      helm template nats ./nats \
+        --namespace ${kubernetes_namespace.nats.metadata[0].name} \
+        --set cluster.enabled=true \
+        --set cluster.replicas=3 \
+        --set nats.jetstream.enabled=true > nats-resources.yaml
+      kubectl apply -f nats-resources.yaml
+    EOT
+  }
+}
+
+
+resource "null_resource" "create_natsschannel_crd" {
+  depends_on = [null_resource.setup_nats]
+  provisioner "local-exec" {
+    command = <<-EOT
+      cat <<EOF | kubectl apply -f -
+      apiVersion: apiextensions.k8s.io/v1
+      kind: CustomResourceDefinition
+      metadata:
+        name: natsschannels.messaging.knative.dev
+      spec:
+        group: messaging.knative.dev
+        names:
+          kind: NatssChannel
+          plural: natsschannels
+          singular: natsschannel
+          listKind: NatssChannelList
+        scope: Namespaced
+        versions:
+        - name: v1alpha1
+          served: true
+          storage: true
+          schema:
+            openAPIV3Schema:
+              type: object
+              properties:
+                spec:
+                  type: object
+                  x-kubernetes-preserve-unknown-fields: true
+      EOF
+    EOT
+  }
+}
+
+
+resource "null_resource" "configure_knative_nats" {
+  provisioner "local-exec" {
+    command = <<-EOT
+      cat <<EOF | kubectl apply -f -
+      ---
+      apiVersion: v1
+      kind: ConfigMap
+      metadata:
+        name: default-ch-webhook
+        namespace: knative-eventing
+      data:
+        default-ch-config: |
+          clusterDefault:
+            apiVersion: messaging.knative.dev/v1alpha1
+            kind: NatssChannel
+      ---
+      apiVersion: v1
+      kind: ConfigMap
+      metadata:
+        name: config-br-defaults
+        namespace: knative-eventing
+      data:
+        default-br-config: |
+          clusterDefault:
+            brokerClass: MTChannelBasedBroker
+            apiVersion: v1
+            kind: ConfigMap
+            name: config-br-default-channel
+            namespace: knative-eventing
+      ---
+      apiVersion: v1
+      kind: ConfigMap
+      metadata:
+        name: config-br-default-channel
+        namespace: knative-eventing
+      data:
+        channelTemplateSpec: |
+          apiVersion: messaging.knative.dev/v1alpha1
+          kind: NatssChannel
```

Comment:
Why is there a second "s"?

Reply:
it's for `s`treaming, it's used to distinguish between other NATS setups, this is the common naming convention in knative it seems